tests/test_api.py

```diff
@@ -333,6 +333,47 @@ def test_retrieve_instance_metadata_wado_prefix(httpserver, client, cache_dir):
     assert request.path == expected_path
 
 
+def test_retrieve_series(client, httpserver, cache_dir):
+    cache_filename = str(cache_dir.joinpath('file.dcm'))
+    with open(cache_filename, 'rb') as f:
+        payload = f.read()
+    content = b''
+    for i in range(3):
+        content += b'\r\n--boundary\r\n'
+        content += b'Content-Type: application/dicom\r\n\r\n'
+        content += payload
+    content += b'\r\n--boundary--'
+    headers = {
+        'content-type': (
+            'multipart/related; '
+            'type="application/dicom"; '
+            'boundary="boundary" '
+        ),
+    }
+    httpserver.serve_content(content=content, code=200, headers=headers)
+    study_instance_uid = '1.2.3'
+    series_instance_uid = '1.2.4'
+    sop_instance_uid = '1.2.5'
+    results = client.retrieve_series(
+        study_instance_uid, series_instance_uid
+    )
+    assert inspect.isgenerator(results)
```

Comment:
Signature and docstring currently only state that the method returns an `Iterator`. Do we need to assert that the returned value is a `Generator`?

Reply:
It's not necessary per se; I avoided type-hinting the methods as generators as I think that can be more complex/confusing than helpful. Anyhow, removed the assertion.